### PR TITLE
Parameterize Drupal version

### PR DESCRIPTION
**Pull request by scottrigby**
See #68.

Giving this a shot before adding other params.

Realized to keep backwards compatibility we need to to set `BITNAMI_IMAGE_VERSION` and `BITNAMI_IMAGE_CHECKSUM` separately.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -17,7 +20,7 @@ RUN bitnami-pkg install mysql-client-10.1.21-0 --checksum 8e868a3e46bfa59f3fb4e1
 RUN bitnami-pkg install drush-8.0.5-1 --checksum cdea2d5067ef67bcda7e5cfe96798c2d6e0167578395fac2d7a5f92a814ecc69
 
 # Install drupal
-RUN bitnami-pkg unpack drupal-8.2.6-0 --checksum a8f874db6aef1e4dae22b2fd88e700b37f405f916ef0e8ad4906f3744e60eec7
+RUN bitnami-pkg unpack drupal-${BITNAMI_IMAGE_VERSION:-8.2.6-r0} --checksum ${BITNAMI_IMAGE_CHECKSUM:-a8f874db6aef1e4dae22b2fd88e700b37f405f916ef0e8ad4906f3744e60eec7}
```
Comment (prydonius) [inline Dockerfile]: this version is slightly different to the BITNAMI_IMAGE_VERSION, it doesn't use the `-rX` suffix, and instead has a different revision `8.2.6-X` where X = 0 in this case. This allows us to keep the BITNAMI_IMAGE_VERSION independent to different package versions and revisions.

**Comment by prydonius** [comment]
@andresmgot @sameersbn what do you think about this parameterised approach to supporting multiple different major versions?

File: Dockerfile (hunk 1 above)
Comment (scottrigby) [inline Dockerfile]: @prydonius I somehow missed that 😄 Changed back to `-rX` pattern in 0a167ea.

**Comment by scottrigby** [comment]
@andresmgot @sameersbn re what @prydonius said, I suggested this strategy because - if there were an easy way to build these images - I could see wider adoption (standalone or via Helm charts). This will be especially valuable now that Drupal 8 has moved to semver (and even still, many sites will still be running Drupal 7 for a long time).

The image build process could be automated with a parameterized template like this + a Jenkins job. New Drupal release notifications could also be connected to Slack or wherever is helpful by pulling from the [feed](https://www.drupal.org/node/3060/release/feed) from [this page](https://www.drupal.org/project/drupal/releases) on d.o. But those could be added in stages, if they seem like useful enough additions.

If this is interesting to you guys, I'm happy to help with some of this, through PRs, Jenkins & Slack webhook setup, etc.

**Comment by sameersbn** [comment]
@scottrigby What is the use case for this PR?

This change would break our CI pipeline and may become difficult to manage if additional variables are parameterized as build time variables. Our internal container update tools would need to be updated to support this.

We are currently working on supporting multiple versions in our repo's and the changes in this PR may not be required if the sole use of these changes are to support building multiple versions.

/cc @prydonius @andresmgot

**Comment by juan131** [comment]
As @sameersbn mentioned we will support multiple versions for some applications as Drupal, Magento, Odoo, etc. in the Docker Images. 

I think it can be a good option to explore for supporting multiple versions. Thanks for the suggestion. However, we can't accept this PR in this moment since it is not aligned with the current CI pipeline.

/cc @aruiz14 @andresmgot 

Regarding the versions tracking, we already have internal tools that help us to obtain the latest versions of every app automatically.

**Comment by scottrigby** [comment]
@sameersbn @juan131 Fair enough, this was an experimental proposal.

However, to your question/point:
> We are currently working on supporting multiple versions in our repo's and the changes in this PR may not be required if the sole use of these changes are to support building multiple versions.

Parameterizing would address both the requests for multiple Drupal versions (without requiring separate branches for each) and multiple PHP versions for each (and allow for future variations). It's generally when multiple params are combined that their value becomes obvious - in this case, one branch for any of the above combinations, rather than either: 1. not supporting all supported PHP versions for all supported Drupal versions (as is the case now), or 2. requiring all these branches:
- Drupal 7.x + PHP 5.6
- Drupal 7.x + PHP 7.1
- Drupal 8.1.x + PHP 5.6
- Drupal 8.1.x + PHP 7.1
- Drupal 8.2.x + PHP 5.6
- Drupal 8.2.x + PHP 7.1
- Drupal 8.3.x + PHP 5.6
- Drupal 8.3.x + PHP 7.1

**Comment by scottrigby** [comment]
Also one last note. Docker Docker [args](https://docs.docker.com/engine/reference/builder/#arg) can have a default, as this PR does (keeping the value you had already) - so it should be backwards-compatible for any current build process you have now – but would also allow passing different versions for beta or alternate workflows, like `docker build --build-arg BITNAMI_IMAGE_VERSION=SOME_DRUPAL_7_VERSION` for a different build process POC. Whatever works for you guys is of course what you should do, but I'm curious how else - or if - you plan to support those version combinations above?

**Comment by scottrigby** [comment]
I see, you're using a version per directory. Not sure how or if you plan to support other options (like PHP version, NGINX or Apache, etc), but I'll hang tight and see 😄 I'll go ahead and close this PR proposal.

**Comment by andresmgot** [comment]
Hi @scottrigby @prydonius @juan131 @sameersbn,

Thank you for your suggestions but I think that build arguments are not a suitable solution for us in order to support different release series for all our applications. Note that in some cases a new major version will require changes in the entrypoint script or new user inputs so we will need to modify the associated files with changes that usually are not backward compatibility. It is true that using build args would cover most of the cases but we need a solution to cover all the possible scenarios. For doing so we chose to go with folders per release series in which we include all the files required to build the image:
https://github.com/bitnami/bitnami-docker-php-fpm/7.1/

For the time being we won't officially support more than one runtime per application because we cannot assure that all the possible combinations between runtime + application would work. We will stick to the most stable recommended release series given by the upstream project:
https://www.drupal.org/docs/7/system-requirements/php#php_required

Note that these decisions may change over the evolution of the project but currently it is the solution that best fits our needs.

**Pull request closed without merging**